pages/content/ch2/propositional-logic.mdx

```diff
@@ -50,41 +50,24 @@ $\vDash$
 
 ## EBNF: Statements vs Formulas
 
-This tries to define formulas and statements with EBNF. Note that all literals are in double quotes.
-
-$$
-\begin{aligned}
-\textit{Symbol} & \Leftarrow \text{A-Z} \\
-\\
-\textit{LogicalOperator} & \Leftarrow \text{"} \land \text{"} \\
-       & \quad \vert \; \text{"} \leftrightarrow \text{"} \\
-       & \quad \vert \; \text{"} \lor \text{"} \\
-       & \quad \vert \; \text{"} \to \text{"} \\
-       & \quad \vert \; \text{"} \lnot \text{"} \\
-\\  
-\textit{FormulaToStatementOperator} & \Leftarrow \text{"} \equiv \text{"} \\
-       & \quad \vert \; \text{"} \models \text{"} \\
-\\
-\textit{StatementOperator} & \Leftarrow \text{"} \implies \text{"} \\
-       & \quad \vert \; \text{"} \iff \text{"} \\
-       & \quad \vert \; \text{"and"} \\
-       & \quad \vert \; \text{"or"} \\
-\\   
-\textit{Formula} & \Leftarrow \textit{Symbol} \\
-       & \quad \vert \; \textit{Symbol } \textit{ LogicalOperator } \textit{ Symbol} \\
-       & \quad \vert \; "("\textit{Formula } \textit{ LogicalOperator } \textit{ Formula}")" \\
-       & \quad \vert \; \text{"} \top \text{"} \\
-       & \quad \vert \; \text{"} \bot \text{"} \\
-\\   
-\textit{Statement} & \Leftarrow \textit{Symbol} \\
-       & \quad \vert \; \textit{Statement } \textit{ StatementOperator } \textit{ Statement} \\
-       & \quad \vert \; \textit{Statement } \text{ "is false"} \\
-       & \quad \vert \; \textit{Formula } \textit{ FormulaToStatementOperator } \textit{ Formula} \\
-       & \quad \vert \; \text{"} \models \text{"} \textit{ Formula} \\
-\end{aligned}
-$$
-
+<div className="mt-4 [&_code]:font-[inherit] [&_code]:italic">
+
+  This tries to define formulas and statements with EBNF. Bold symbols are used to denote EBNF syntax elements **(** **)** **\[** **]** **│**. References to existing rules are made using `Rulename…` element.
+
+  | Rule                | Expression                                                                                                 |
+  |---------------------|------------------------------------------------------------------------------------------------------------|
+  | `PropSymbol`        | $A$ **│** $B$ **│** $C$ **│** ••• **│** $Z$                                                                |
+  | `Symbol`            | `PropSymbol` **│** $\top$ **│** $\bot$                                                                     |
+  | `LogicOperator`     | $\land$ **│** $\lor$ **│** $\leftrightarrow$ **│** $\to$                                                   |
+  | `Negation`          | $\lnot$                                                                                                    |
+  | `Combination`       | **(**`Formula` **│** $($`Combination`$)$**)** `Operator` **(**`Formula` **│** $($`Combination`$)$**)**     |
+  | `Formula`           | **[**`Negation`**]**`Symbol` **│** `Combination` **│** `Negation`$($`Combination`$)$                       |
```

Comment:
I don't think that we need `Combination`, we could do the following instead:
```
`Symbol` **|** **[**`Negation`**]** `Formula` **|** `Formula` `Operator` `Formula` **|** (`Formula`)
```
A formula is either a symbol, a negated formula or the output of an operation on two formulas (semantics of prop. logic)

Reply:
I thought this way it is more clear, that combination of combinations needs round brackets. 
E.g. $A ∧ (B ∨ C) ≢  (A ∧ B) ∨ C$ and $¬A ∧ B ≢  ¬(A ∧ B)$

Reply:
Ah ok, this makes sense (and now I don't see how one could avoid a second ebnf "type")

Reply:
But you are right, this way $A \lor B \lor C$ would not be valid 😕